nbconvert/filters/markdown_mistune.py

```diff
@@ -102,7 +107,11 @@ def block_code(self, code, lang):
 
     def header(self, text, level, raw=None):
         html = super(IPythonRenderer, self).header(text, level, raw=raw)
-        return add_anchor(html)
+        if hasattr(self, 'local_config'):
```

Comment:
Using `hasattr` where the code already knows what the object is is a code smell, in my opinion. In this case, I'd instantiate `IPythonRenderer` with a config object that can be passed down. If we want it to be optional, the default can be an empty Config instance.